Basic.Azure.Storage/Communications/ServiceExceptions/AzureException.cs

```diff
@@ -1,28 +1,30 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Net;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace Basic.Azure.Storage.Communications.ServiceExceptions
-{
-    public class AzureException : Exception
-    {
-        public string RequestId { get; private set; }
-        public HttpStatusCode StatusCode { get; private set; }
-        public string StatusDescription { get; private set; }
-        public Dictionary<string, string> Details { get; private set; }
-
-        public AzureException(string requestId, HttpStatusCode statusCode, string statusDescription, Dictionary<string, string> details, WebException baseException)
-            : base(String.Format("{0}. Request {1}, Http Status: {2}\n{3}", statusDescription, requestId, (int)statusCode, string.Join(", ", details.Select(kvp => String.Format("{0}: {1}", kvp.Key, kvp.Value)))), baseException)
-        {
-            RequestId = requestId;
-            StatusCode = statusCode;
-            StatusDescription = statusDescription;
-            Details = details;
-        }
-
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+
+namespace Basic.Azure.Storage.Communications.ServiceExceptions
+{
+    public class AzureException : Exception
+    {
+        public string RequestId { get; private set; }
+        public HttpStatusCode StatusCode { get; private set; }
+        public string StatusDescription { get; private set; }
+        public Dictionary<string, string> Details { get; private set; }
+        public Stack<Exception> ExceptionRetryStack { get; internal set; }

```

Comment:
Internal because, when retrying above, the exception will have already been thrown (so we can't set the retryStack through the constructor), but I don't want to expose this as a settable property. Thoughts?